bpf/lib/nodeport.h

```diff
@@ -194,39 +198,125 @@ static __always_inline bool nodeport_uses_dsr6(const struct ipv6_ct_tuple *tuple
 static __always_inline bool snat_v6_needed(struct __ctx_buff *ctx,
 					   union v6addr *addr)
 {
+	union v6addr masq_addr __maybe_unused;
+	union v6addr dr_addr = IPV6_DIRECT_ROUTING;
+	struct remote_endpoint_info *remote_ep __maybe_unused;
+	struct endpoint_info *local_ep __maybe_unused;
 	void *data, *data_end;
 	struct ipv6hdr *ip6;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip6))
 		return false;
-#ifdef ENABLE_DSR_HYBRID
+
+	/* We need to SNAT when there is a potential of overlapping tuples, e.g.
```

Comment:
Nit: not sure what is a convention here, but I usually try to avoid duplicating v4/v6 comments, and instead I comment only the v4 equivalent. Otherwise, we are risking that comments might get stale.